plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -1077,6 +1078,22 @@ protected void verifyColumnName(DatabaseMetaData databaseMetadata, String column
         }
     }
 
+    @Override
+    public OptionalInt getMaxColumnNameLength(ConnectorSession session)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
```

Comment:
The PostgreSQL driver has some connection pool by default, right? Otherwise, this would be very costly.

Reply:
Yes, see the comment https://github.com/trinodb/trino/pull/18984#discussion_r1322421663

This is open to being changed. I don't know what's the ideal solution - looking for input.